src/icon/editIcon.tsx

```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+
+import { color } from '../_utils/branding'
+import { BaseIcon, BaseIconDefaultProps, Icon } from '../_utils/icon'
+
+export const EditIcon = (props: Icon) => (
+  <BaseIcon {...props}>
+    <path
+      fillRule="evenodd"
+      clipRule="evenodd"
+      d="M18.0678 2.12264C17.8726 1.92738 17.556 1.92738 17.3607 2.12264L3.07502 16.4084C3.00522 16.4782 2.95764 16.5671 2.93828 16.6639L1.9859 21.4258C1.95311 21.5897 2.00442 21.7592 2.12264 21.8774C2.24085 21.9956 2.41032 22.0469 2.57425 22.0141L7.33615 21.0617C7.43295 21.0424 7.52185 20.9948 7.59165 20.925L18.9969 9.51974C19.0055 9.51224 19.0139 9.5044 19.0221 9.49621C19.0303 9.48802 19.0381 9.47962 19.0456 9.47102L21.8774 6.63927C21.9711 6.54551 22.0238 6.41833 22.0238 6.28572C22.0238 6.15311 21.9711 6.02593 21.8774 5.93217L18.0678 2.12264ZM18.6677 8.43472L20.8167 6.28572L17.7143 3.1833L15.5653 5.3323L18.6677 8.43472ZM14.8582 6.03941L3.88917 17.0084L3.11357 20.8864L6.99159 20.1108L17.9606 9.14183L14.8582 6.03941Z"
+      fill={props.isDisabled ? color.gray : props.iconColor}
```

Comment:
please avoid applying the style on the SVG itself since its a strong selector weight and high on specificity.
First of all it disabled behaviorshould be a common ground for any Icon on the library, second would be better to use css (-in-js) instead then.